presto-main/src/main/java/com/facebook/presto/server/security/CustomPrestoAuthenticator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import com.facebook.airlift.http.server.AuthenticationException;
+import com.facebook.airlift.http.server.Authenticator;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import java.security.Principal;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static java.util.Collections.list;
+import static java.util.Objects.requireNonNull;
+
+public class CustomPrestoAuthenticator
+        implements Authenticator
+{
+    private PrestoAuthenticatorManager authenticatorManager;
+
+    @Inject
+    public CustomPrestoAuthenticator(PrestoAuthenticatorManager authenticatorManager)
+    {
+        this.authenticatorManager = requireNonNull(authenticatorManager, "authenticatorManager is null");
+    }
+
+    @Override
+    public Principal authenticate(HttpServletRequest request)
+            throws AuthenticationException
+    {
+            // Extracting headers into a Map
```

Comment:
We will need to catch AccessDeniedException here and throw it as AuthenticationException, we cannot remove that.

Reply:
We can remove the catch clause for RuntimeException but we need to catch AccessDeniedException.